optimum_benchmark/backends/base.py

```diff
@@ -62,18 +67,28 @@ class BackendConfig(ABC):
     # clean up options
     delete_cache: bool = False
 
+    def __post_init__(self):
+        if self.inter_op_num_threads is not None:
+            if self.inter_op_num_threads == -1:
+                self.inter_op_num_threads = cpu_count()
+
+        if self.intra_op_num_threads is not None:
+            if self.intra_op_num_threads == -1:
+                self.intra_op_num_threads = cpu_count()
+
 
 class Backend(ABC):
-    # model and pipeline benchmarks
-    pretrained_model: Union[PreTrainedModel, Pipeline]
-    # only for model benchmarks
-    pretrained_config: Optional[PretrainedConfig]
+    name: str
+    config: ClassVar[BackendConfig]
+
+    pretrained_model: Union[PreTrainedModel, DiffusionPipeline]
     pretrained_processor: Optional[PreTrainedProcessor]
+    pretrained_config: Optional[PretrainedConfig]
 
-    def __init__(self, model: str, task: str, device: str, hub_kwargs: DictConfig):
+    def __init__(self, model: str, task: str, device: str, hub_kwargs: Dict[str, Any]):
         self.model = model
         self.task = task
-        self.device = torch.device(device)
+        self.device = device
```

Comment:
why?

Reply:
in an attempt to reduce torch usage outside of its backend. 
I'll revert it since I already found a way to not import torch before `CUDA_VISIBLE_DEVICES` is set.

Reply:
makes sense